src/controllers/products/createProduct.ts

```diff
@@ -0,0 +1,159 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { getDb } from '../../utils/db';
+import type { Product } from '../../models/product';
+import { AuditLogAction } from '../../models/auditLog';
+import { updateAuditLog } from '../../utils/auditLog';
+import { ObjectId } from 'mongodb';
+import { ResponseWrapper } from '../../utils/responseWrapper';
+import { verifyJWT } from '../../utils/authUtils';
+
+/**
+ * Event doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+ * @param {Object} event - API Gateway Lambda Proxy Input Format
+ *
+ * Return doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html
+ * @returns {Object} object - API Gateway Lambda Proxy Output Format
+ *
+ */
+
+export const lambdaHandler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+    try {
+        if (!event.body) {
+            return ResponseWrapper.badRequest('Request body is required');
+        }
+
+        const authHeader = event.headers?.Authorization || event.headers?.authorization;
+        if (!authHeader) {
+            return ResponseWrapper.unauthorized('Unauthorized');
+        }
+
+        const token = authHeader.split(' ')[1];
+
+        // Check if the user is valid - both users and admins can create products
+        const { isValid, payload } = await verifyJWT(token);
+        if (!isValid) {
+            return ResponseWrapper.unauthorized('Unauthorized');
+        }
+
+        let input: Product;
+        try {
+            input = JSON.parse(event.body);
+        } catch (error) {
+            return ResponseWrapper.badRequest('Invalid JSON in request body');
+        }
+
+        // Validate required fields including status and master_version
+        if (
```

Comment:
This was in my PR too, we can think of a more elegant way to handle and address this in upcoming PRs